funcframework/framework.go

```diff
@@ -273,6 +273,10 @@ func runUserFunctionWithContext(ctx context.Context, w http.ResponseWriter, r *h
 }
 
 func writeHTTPErrorResponse(w http.ResponseWriter, statusCode int, status, msg string) {
+	// Ensure logs end with a newline otherwise they are grouped incorrectly in SD.
+	if msg!=nil && msg != "" && !strings.HasSuffix(msg, "\n") {
```

Comment:
Don't need this !=nil check for a string.

Reply:
... and the msg!="" is also not needed.

Reply:
or maybe you want
```
if msg == "" {
  return
}
```

I'm not sure what is intended.

Reply:
Just a momentary glitch in my matrix.